Table/src/lib/FitsReaderHelper.cpp

```diff
@@ -62,51 +62,72 @@ std::type_index asciiFormatToType(const std::string& format) {
-  } else if (format[0] == 'B') {
-    return typeid(int32_t);
-  } else if (format[0] == 'I') {
-    return typeid(int32_t);
-  } else if (format[0] == 'J') {
-    return typeid(int32_t);
-  } else if (format[0] == 'K') {
-    return typeid(int64_t);
-  } else if (format.back() == 'A') {
-    return typeid(std::string);
-  } else if (format[0] == 'E') {
-    return typeid(float);
-  } else if (format[0] == 'D') {
-    return typeid(double);
-  } else if (shape.empty()) {
-    if (format.back() == 'B') {
-      return typeid(std::vector<int32_t>);
-    } else if (format.back() == 'I') {
-      return typeid(std::vector<int32_t>);
-    } else if (format.back() == 'J') {
-      return typeid(std::vector<int32_t>);
-    } else if (format.back() == 'K') {
-      return typeid(std::vector<int64_t>);
-    } else if (format.back() == 'E') {
-      return typeid(std::vector<float>);
-    } else if (format.back() == 'D') {
-      return typeid(std::vector<double>);
-    }
-  } else {
-    if (format.back() == 'B') {
+  // Get the size out of the format string
+  char ft = format.front();
+  int size = 1;
+  if (std::isdigit(format.front())) {
+    size = std::stoi(format.substr(0, format.size() - 1));
+    ft = format.back();
+  }
+
+  // If shape is set, it is an NdArray
+  if (!shape.empty()) {
+    if (ft == 'B') {
       return typeid(NdArray<int32_t>);
-    } else if (format.back() == 'I') {
+    } else if (ft == 'I') {
       return typeid(NdArray<int32_t>);
-    } else if (format.back() == 'J') {
+    } else if (ft == 'J') {
       return typeid(NdArray<int32_t>);
-    } else if (format.back() == 'K') {
+    } else if (ft == 'K') {
       return typeid(NdArray<int64_t>);
-    } else if (format.back() == 'E') {
+    } else if (ft == 'E') {
       return typeid(NdArray<float>);
-    } else if (format.back() == 'D') {
+    } else if (ft == 'D') {
       return typeid(NdArray<double>);
     }
   }
+  // If the dimensionality is 1, it is a scalar
+  else if (size == 1) {
+    if (ft == 'L') {
+      return typeid(bool);
+    }
+    else if (ft == 'B') {
+      return typeid(int32_t);
+    }
+    else if (ft == 'I') {
+      return typeid(int32_t);
+    }
+    else if (ft == 'J') {
+      return typeid(int32_t);
+    }
+    else if (ft == 'K') {
+      return typeid(int64_t);
+    }
+    else if (ft == 'E') {
+      return typeid(float);
+    }
+    else if (ft == 'D') {
+      return typeid(double);
+    }
+  }
+  // Last, vectors
+  else {
+    if (ft == 'B') {
+      return typeid(std::vector<int32_t>);
+    } else if (ft == 'I') {
+      return typeid(std::vector<int32_t>);
+    } else if (ft == 'J') {
+      return typeid(std::vector<int32_t>);
+    } else if (ft == 'K') {
+      return typeid(std::vector<int64_t>);
+    } else if (ft == 'E') {
+      return typeid(std::vector<float>);
+    } else if (ft == 'D') {
+      return typeid(std::vector<double>);
+    } else if (ft == 'A') {
+      return typeid(std::string);
+    }
+  }
```

Comment:
I wanted to refactor this part since using "switch" or "if then else if else if else if ..." is considered bad style. People recommand to replace these structures with map constants. But we can leave it like this for now.

Reply:
Thanks for the approval. You are right, a map or something of the sort would look nicer than a bunch of if/elses. I'll try to remember when (if) I have to touch this again.